### PR TITLE
Allow passing in `$popertyId` dynamically

### PR DESCRIPTION
**Pull request by alexacron**
The optional parameter $propertyId allows bypassing the configured parameter. This allows adding properties via API then using them here. Useful for cross-domain tracking also.

### CODE REVIEW

**Comment by Plytas** [comment]
Thanks for the contribution @alexacron. Would you mind writing tests for this?

File: src/Analytics.php [hunk 1]
```diff
@@ -26,9 +26,11 @@ class Analytics
     /**
      * @throws InvalidPropertyIdException
      */
-    public function __construct()
+    public function __construct(?string $propertyId = null)
     {
-        $propertyId = config('analytics.property_id');
+        if ($propertyId == null) {
+            $propertyId = config('analytics.property_id');
+        }
```
Comment (Plytas) [inline src/Analytics.php:33]:
```suggestion
        $propertyId ??= config('analytics.property_id');
```

File: src/Analytics.php [hunk 2]
```diff
@@ -38,10 +40,10 @@ public function __construct()
         $this->requestData = new RequestData(propertyId: $propertyId);
     }
 
-    public static function query(): static
+    public static function query(?string $propertyId=null): static
```
Comment (Plytas) [inline src/Analytics.php:43]:
```suggestion
    public static function query(?string $propertyId = null): static
```

**Review by Plytas — CHANGES_REQUESTED** [review]
@alexacron couple of syntactic changes, but otherwise looks good!

**Comment by Plytas** [comment]
@gtmassey Could you look into failing pint action?

**Review by alexacron — COMMENTED** [review]
I've run pint locally and committed the changes. I don't know why they fail on remote on the same files I just fixed.

**Comment by Plytas** [comment]
It's the commit step of the action that fails due to permission issue.

**Comment by alexacron** [comment]
> It's the commit step of the action that fails due to permission issue.

Anything I can do ?

**Comment by Plytas** [comment]
@alexacron would you mind giving me write permission to your fork? Without it I cannot push changes to this PR. I want to bump composer dev dependencies and also bump Pint action version so it can use newest Pint. For some reason the old version still picks up changes even though you already committed them.

**Comment by alexacron** [comment]
I sent you an invitation

**Review by Plytas — APPROVED** [review]

**Pull request merged**